scripts/helpers.js

```diff
@@ -18,6 +19,31 @@ try {
 } catch (e) {}
 
 /**
+ * Generate data structure for generating table of contents.
+ *
+ * @param {string} page - Page content.
+ * @returns {array} [{title: 'Title', link: '#anchor', children: []}]
+ */
+hexo.extend.helper.register('table_of_contents', function (page) {
+  var $ = htmlParser.load(page);
+  var h2Set = $('h2');
+  var toc = [];
+
+  for (var i = 0 ; i < h2Set.length; i++) {
+    var h2 = h2Set[i];
+    var link = h2.children[0];
+    var item = {
+      title: link.attribs.title,
+      href: link.attribs.href,
+      children: []
+    };
+    toc.push(item);
```

Comment:
anything wrong with just pushing the whole `link` object? too much stuff in there?